modules/integration/tests-integration/tests/src/test/java/org/wso2/carbon/integration/test/client/Wso2EventServer.java

```
@@ -138,23 +135,26 @@ public void receive(List<Event> eventList, Credentials credentials) {
 
         });
         thriftDataReceiver.start("0.0.0.0");
-        System.out.println("Test Server Started");
-
+        log.info("Test Server Started.");
     }
 
     public void stop() {
-        Assert.assertTrue(eventReceived);
-        thriftDataReceiver.stop();
-        System.out.println("Test Server Stopped");
+        if (!eventReceived) {
+            log.warn("Events did not received.");
+        }
+        try {
+            thriftDataReceiver.stop();
+            log.info("Test Server Stopped.");
+        } catch (Throwable t) {
```

Here also, let any exceptions be thrown, and handle it at the top of the stack.